In [2]:
import pandas as pd
import socnet as sn
import json
import scipy
import numpy as np
from itertools import permutations
import networkx as nx
import seaborn as sns
import matplotlib.pyplot as plt

fp = open('clean_data.json', 'r')
data = json.load(fp)

In [7]:
print(len(data))

144


Para gerar todos os nós, é preciso fazer uma lista com todos os jogadores da rede. Nosso scrape só varreu o perfil de 144 pessoas.

In [6]:
all_players = []
for player in data:
    all_players.append(player)
    for team_mate in data[player]["played_with"]:
        all_players.append(team_mate)
all_players = list(set(all_players)) #remove jogadores repetidos
print(len(all_players))

2836


In [3]:
def create_gml():
    gml = open("players.gml", "w")
    gml.write("graph [\n  directed 1\n  weighted 1\n")
    for player in all_players:
        gml.write('  node [\n    id {}\n    label "{}"\n  ]\n'.format(player, str(player)))
    for player in data:
        for team_mate, times_played in data[player]["played_with"].items():
            gml.write("  edge [\n    source {}\n    target {}\n    weight {}\n  ]\n".format(player, team_mate, times_played))
    gml.write("]")
    gml.close()
create_gml()

In [4]:
sn.node_size = 10
sn.node_color = (255, 255, 255)

sn.edge_width = 1
sn.edge_color = (192, 192, 192)

sn.node_label_position = 'hover'
g = sn.load_graph('players.gml', has_pos=False)
sn.show_graph(g, nlab=True)